WooCommerce/Classes/ViewModels/CardPresentPayments/PaymentCaptureOrchestrator.swift

```diff
@@ -234,7 +226,7 @@ private extension PaymentCaptureOrchestrator {
                            onCompletion: @escaping ((Result<PaymentParameters, Error>) -> Void)) {
         guard let orderTotal = currencyFormatter.convertToDecimal(from: order.total) else {
             DDLogError("Error: attempted to collect payment for an order without a valid total.")
-            onCompletion(Result.failure(NotValidAmountError.other))
+            onCompletion(Result.failure(CollectOrderPaymentUseCase.NotValidAmountError.other))
```

Comment:
The `NotValidAmountError` is moved to `CollectOrderPaymentUseCase`. We have to fail here because it's necessary to convert the order total to decimal.

Reply:
@joshheald That's a great point. I was considering it, but first I hesitated because we would have two sources of truth (Order property and this new parameter, and it could add (another) parameter to the function.
However, it is true that it is good to only convert it once, and remove the errors propagation from the orchestrator, so I changed it in 
https://github.com/woocommerce/woocommerce-ios/pull/6755/commits/943a9d144b4f48cd1ff9d31203f89e8d192d7fc5 and https://github.com/woocommerce/woocommerce-ios/pull/6755/commits/0bc739fa1a5799c0d13d78f52e22b4b5906ff128 . Let me know your thoughts :)

Reply:
@toupper I definitely understand your hesitation on having two places where the order total is, but I think your changes here are an improvement. Making errors impossible simplifies the code and is a big win, and worth this trade off IMO. 

How do you feel about it now you've done it, do you still have these concerns?

I think that if you still don't like it, the way to resolve the duplication would be to make a struct which contains the parts of the order `paymentParameters()` actually needs, and pass that instead of a whole `Order`. But while I think that would be better, it's probably not the most impactful thing to do right now.

Reply:
I agree, it is a good trade-off for sure :) I like what we have now, even if there are few parameters it is still a clean interface. I will leave it as it is for now, and if we add another parameter we go with the `struct` approach you just described 👍